netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -138,6 +140,11 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
 
       SSLEngine sslEngine = sslContext.newEngine(ctx.alloc());
       ctx.pipeline().addFirst(new SslHandler(sslEngine, false));
+
+      // TODO(lukaszx0) Seems like there might be a better way to pass this object other than
```

Comment:
This doesn't bother me too much, but I think we'd eventually want to plumb it through ProtocolNegotiator.Handler and pass the handler into NettyClientHandler (since it will eventually want something like this) and NettyServerHandler. That looks relatively easy.